src/close.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{transaction::CloseTransaction, Channel};
+use bitcoin::{Address, Transaction};
+use ecdsa_fun::Signature;
+
+pub struct State0(Channel);
+pub struct State1 {
+    channel: Channel,
+    final_address_self: Address,
+    final_address_other: Address,
+}
+pub struct FinalState(Transaction);
+
+impl FinalState {
+    pub fn into_transaction(self) -> Transaction {
+        self.0
+    }
+}
+
+impl From<Channel> for State0 {
+    fn from(channel: Channel) -> Self {
+        Self(channel)
+    }
+}
+
+pub struct Message0 {
+    final_address: Address,
+}
+
+#[derive(Debug)]
+pub struct Message1 {
+    close_transaction: CloseTransaction,
+    sig_close_transaction: Signature,
+}
+
+impl State0 {
+    pub fn compose(&self, final_address: Address) -> Message0 {
+        Message0 { final_address }
+    }
+
+    pub fn interpret(
+        self,
+        final_address_self: Address,
```

Comment:
:mega: I'm not so fond of this. I would say it would be better to first instantiate `State0` with `Channel` _and_ `final_address_self` and then compose and interpret without the extra argument. That way you ensure that you don't call these functions with a different `final_address_self` by mistake.